### PR TITLE
Embed JS: fix incompatibilities with sphinx 6.x (jquery removal)

### PR DESCRIPTION
**Pull request by stsewd**
- $.urlencode was an alias for `encodeURIComponent`, use that instead.
  https://github.com/sphinx-doc/sphinx/blob/b26bf0320cf3ec8a3400a70b2dc481d511ff2563/doc/_themes/sphinx13/static/_sphinx_javascript_frameworks_compat.js#L33-L33
- We were using underscore.js just for the template functionality, so I just replaced that with the DOM API.
- JQuery isn't included by default on sphinx, so we are injecting it if isn't present. We should migrate all of our code to be jquery free, but this is a first step.

Our theme still doesn't work becuase of https://github.com/readthedocs/sphinx_rtd_theme/pull/1299.
But this PR doesn't depend on the theme, it can be merged as is.

ref https://github.com/sphinx-doc/sphinx/issues/10574

### CODE REVIEW

File: readthedocs/core/static-src/core/js/doc-embed/search.js
Comment (humitos) [inline readthedocs/core/static-src/core/js/doc-embed/search.js:80]: I took a quick look at this code and it looks correct to me, but I'm not 100% sure. I don't know a good way to try this PR and QA the code. I think we don't have tests for this, either.

Probably it is worth having a second pair of eyes on the review checking the translation from the template used to this code.

Also, it would be good to have a longer comment explaining what this does exactly and examples of the HTML code generated by this.

File: readthedocs/core/static-src/core/js/readthedocs-doc-embed.js
Comment (humitos) [inline readthedocs/core/static-src/core/js/readthedocs-doc-embed.js:22]: We should update the comment from line 10-11 of this file too.

File: readthedocs/core/static-src/core/js/readthedocs-doc-embed.js
Comment (humitos) [inline readthedocs/core/static-src/core/js/readthedocs-doc-embed.js:30]: What's up if this line is executed first and then jQuery is loaded by a different extension/theme/etc?

I'm thinking about this scenario;

1. this code is run and we inject jQuery
2. another .js file is loaded that includes jQuery
3. once `domReady` is called, we will have 2 different jQuery versions/files loaded

Shouldn't we inject jQuery synchronously inside `domReady` so we wait for all the scripts to be loaded first and avoid this scenario?

**Review by humitos — COMMENTED** [review]
I think this code is fine. However, if I understand correctly, it seems there are some edge cases we won't be handling that I suppose will break in production due to the order of the scripts loaded.

File: readthedocs/core/static-src/core/js/readthedocs-doc-embed.js
Comment (humitos) [inline readthedocs/core/static-src/core/js/readthedocs-doc-embed.js:33]:
```suggestion
      // Inject jQuery if isn't present already.
      if (!window.jQuery) {
          console.log("jQuery not found. Injecting.");
```

**Comment by mgeier** [comment]
Is it really necessary to inject jQuery at this point?

Sphinx still has jQuery, it is only planned be removed in Sphinx 6.0: https://github.com/sphinx-doc/sphinx/issues/10070

Note that https://www.sphinx-doc.org/ uses the `master` branch, with jQuery already removed, but they have added the JS frameworks to their docs already: https://github.com/sphinx-doc/sphinx/blob/master/doc/conf.py#L181-L186, so it shouldn't be necessary to add it from the RTD side.

**Comment by stsewd** [comment]
> Is it really necessary to inject jQuery at this point?

There are some themes on mkdocs that don't include jQuery (material).

**Comment by stsewd** [comment]
I have updated this PR to not inject jquery yet, so isn't blocked on that, this still removes some dependency on jquery.

**Review by agjohnson — APPROVED** [review]
This is a good refactor, it makes a lot of sense to chip away at jQuery removal like this

**Pull request merged**
